File: fix_locales.js
const fs = require('fs');
const path = require('path');

const files = [
  'commands/lifetime-premium.js',
  'commands/premium-role.js',
  'commands/betatester.js'
];

files.forEach(filepath => {
  const fullPath = path.join(__dirname, filepath);
  let content = fs.readFileSync(fullPath, 'utf8');

  // Remove all lines containing 'he:' or 'ar:' as locale keys
  const lines = content.split('\n');
  const filtered = lines.filter(line => {
    const trimmed = line.trim();
    return !trimmed.startsWith('he:') && !trimmed.startsWith('ar:');
  });

  // Fix trailing commas
  const result = [];
  for (let i = 0; i < filtered.length; i++) {
    const line = filtered[i];
    const nextLine = filtered[i + 1];

    // If this line ends with comma and next line starts with }),
    // remove the comma
    if (line.trim().endsWith(',') && nextLine && nextLine.trim().startsWith('})')) {
      result.push(line.replace(/,\s*$/, ''));
    } else {
      result.push(line);
    }
  }

  fs.writeFileSync(fullPath, result.join('\n'), 'utf8');
  console.log(`✅ Cleaned ${filepath}`);
});

console.log('✅ All files cleaned successfully!');

File: fix_locales.py
#!/usr/bin/env python3
import re

files = [
    'commands/lifetime-premium.js',
    'commands/premium-role.js',
    'commands/betatester.js'
]

for filepath in files:
    with open(filepath, 'r', encoding='utf-8') as f:
        content = f.read()

    # Remove lines with he: or ar: locale codes (including the comma from previous line if needed)
    # Pattern: remove lines that are just whitespace + (he|ar): + anything + optional comma
    lines = content.split('\n')
    filtered_lines = []

    for i, line in enumerate(lines):
        # Skip lines that contain 'he:' or 'ar:' as locale keys
        if re.match(r'\s+(he|ar):\s+[\'"]', line):
            # Also remove trailing comma from previous line if it exists
            if filtered_lines and filtered_lines[-1].rstrip().endswith(','):
                # Check if next line doesn't start with a locale key (meaning this was the last locale)
                if i + 1 < len(lines):
                    next_line = lines[i + 1].strip()
                    if next_line.startswith('}'):
                        # Remove comma from last line
                        filtered_lines[-1] = filtered_lines[-1].rstrip().rstrip(',') + '\n' if filtered_lines[-1].endswith('\n') else filtered_lines[-1].rstrip().rstrip(',')
            continue
        filtered_lines.append(line)

    cleaned_content = '\n'.join(filtered_lines)

    with open(filepath, 'w', encoding='utf-8') as f:
        f.write(cleaned_content)

    print(f'✅ Cleaned {filepath}')

print('✅ All files cleaned successfully!')
